observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,194 @@
+
+  private static final MethodFilterParams NO_FILTER_PARAMS
+      = new MethodFilterParams(false, 0, 0);
+
+  private final ObservabilityConfig config;
+  boolean methodOrServiceFilterPresent;
+  boolean globalLog;
+  MethodFilterParams globalParams;
+  Map<String, MethodFilterParams> perServiceFilters;
+  Map<String, MethodFilterParams> perMethodFilters;
+  Set<EventType> logEventTypeSet;
+
+  @VisibleForTesting
+  ConfigFilterHelper(ObservabilityConfig config) {
+    this.config = config;
+  }
+
+  /**
+   * Creates and returns helper instance for log filtering.
+   *
+   * @param config processed ObservabilityConfig object
+   * @return helper instance for filtering
+   */
+  public static ConfigFilterHelper getInstance(ObservabilityConfig config) {
+    ConfigFilterHelper filterHelper = new ConfigFilterHelper(config);
+    filterHelper.setEmptyFilter();
+    if (config.isEnableCloudLogging()) {
+      filterHelper.setMethodOrServiceFilterMaps();
+      filterHelper.setEventFilterSet();
+    }
+    return filterHelper;
+  }
+
+  @VisibleForTesting
+  void setEmptyFilter() {
+    this.globalLog = false;
+    this.globalParams = NO_FILTER_PARAMS;
+    this.methodOrServiceFilterPresent = false;
+    this.perServiceFilters = new HashMap<>();
+    this.perMethodFilters = new HashMap<>();
+    this.logEventTypeSet = ImmutableSet.of();
+    return;
+  }
+
+  @VisibleForTesting
+  void setMethodOrServiceFilterMaps() {
+    ImmutableList<LogFilter> logFilters = config.getLogFilters();
+    if (logFilters == null) {
+      return;
+    }
+
+    boolean globalLog = false;
+    MethodFilterParams globalParams = null;
+    Map<String, MethodFilterParams> perServiceFilters = new HashMap<>();
+    Map<String, MethodFilterParams> perMethodFilters = new HashMap<>();
+
+    UnmodifiableListIterator<LogFilter> logFilterIterator = logFilters.listIterator();
+    while (logFilterIterator.hasNext()) {
+      // '*' for global, 'service/*' for service glob, or 'service/method' for fully qualified
+      LogFilter currentFilter = logFilterIterator.next();
+      String methodOrServicePattern = currentFilter.pattern;
+      int currentHeaderBytes
+          = currentFilter.headerBytes != null ? currentFilter.headerBytes : 0;
+      int currentMessageBytes
+          = currentFilter.messageBytes != null ? currentFilter.messageBytes : 0;
+      if (methodOrServicePattern.equals("*")) {
+        // parse config for global, e.g. "*"
+        globalLog = true;
+        globalParams = new MethodFilterParams(true,
+            currentHeaderBytes, currentMessageBytes);
+      } else if (isServiceGlob(methodOrServicePattern)) {
+        // parse config for a service, e.g. "service/*"
+        String service = MethodDescriptor.extractFullServiceName(methodOrServicePattern);
+        if (perServiceFilters.containsKey(service)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0)", methodOrServicePattern);
+          continue;
```

Comment:
re my comment for line 101: in this case you generate a warning and continue which means all the later duplicate entries are ignored (the header and message bytes values could be different) but for "*" there is no such check (so the later one overrides previous one)